File: sieve/sieve.lua
#!/usr/bin/env lua5.4
local function sieve_sundaram(n)
   local a = {}
   local s = {2}
   local m = n/2
   for i = 1, n do
      for j = i, n do
         local p = i + j + 2*i*j
         if p <= n then
            a[p] = true
         end
      end
   end
   for k = 1, m do
      if not a[k] then
         local q = 2*k + 1
         table.insert(s, q)
      end
   end

   return s
end

local t = sieve_sundaram(1000)
local ts = ""
for _,v in ipairs(t) do
   ts = ts .. v .. " "
end
print(ts)

File: sieve/sieve.py
#!/usr/bin/env python3

def sieve_sundaram(n):
    m,s = n // 2, [2]
    a = [x for i in range(1, n) for j in range(i, n) if (x:=i+j+2*i*j) <= n]
    a = {p:True for p in a}
    s.extend([2 * k + 1 for k in range(1, m+1) if k not in a])
    return s

[print(f"{v} ", end='') for v in sieve_sundaram(1000)]
print('')
